File: ocaml/006/main.ml
(*
    ProjectEuler 006
    OCaml / main.ml
    copyright (c) 2015 Susisu

    OCaml 4.02.1
    `ocamlc main.ml -o bin/main`
*)

let rec calc n s t =
    if n <= 100 then
        calc (n + 1) (s + n) (t + n * n)
    else
        s * s - t;;

let main () = begin
    print_int (calc 1 0 0);
    print_newline ()
end;;

main ();;

File: lua/006/main.lua
--[[
    ProjectEuler 006
    Lua / main.lua
    copyright (c) 2015 Susisu

    Lua 5.1.5
    `lua main.lua`
]]

function main()
    local s = 0
    local t = 0
    for i = 1, 100 do
        s = s + i * i
        t = t + i
    end
    t = t * t
    print(t - s)
end

main()
